From 195965c7a14f8dff8e6a82e5f63a3fb2e8bf5391 Mon Sep 17 00:00:00 2001
From: Peter Ujfalusi <peter.ujfalusi@ti.com>
Date: Fri, 7 Nov 2014 14:37:41 +0200
Subject: [PATCH 1364/1587] ALSA: core: Add module param array to exclude card
 slots from automatic assignment

Via slots_reserved module param array one can specify which card slots
should not be considered when assigning card slots automatically.
These reserved slots can only be taken if a card driver specifically
requests the slot via snd_card_create().

Signed-off-by: Peter Ujfalusi <peter.ujfalusi@ti.com>
[zou: Original patch taken from
ti-sdk-am335x-evm-08.00.00.00-Linux-x86-Install.bin]
Signed-off-by: Cao Zou <cao.zou@windriver.com>
---
 sound/core/init.c |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/sound/core/init.c b/sound/core/init.c
index 39c25167..e41ab69 100644
--- a/sound/core/init.c
+++ b/sound/core/init.c
@@ -57,6 +57,10 @@ static char *slots[SNDRV_CARDS];
 module_param_array(slots, charp, NULL, 0444);
 MODULE_PARM_DESC(slots, "Module names assigned to the slots.");
 
+static bool slots_reserved[SNDRV_CARDS];
+module_param_array(slots_reserved, bool, NULL, 0444);
+MODULE_PARM_DESC(slots_reserved, "Slots which are excluded from automatic assignment");
+
 /* return non-zero if the given index is reserved for the given
  * module via slots option
  */
@@ -148,7 +152,7 @@ static int get_slot_from_bitmask(int mask, int (*check)(struct module *, int),
 	for (slot = 0; slot < SNDRV_CARDS; slot++) {
 		if (slot < 32 && !(mask & (1U << slot)))
 			continue;
-		if (!test_bit(slot, snd_cards_lock)) {
+		if (!test_bit(slot, snd_cards_lock) && !slots_reserved[slot]) {
 			if (check(module, slot))
 				return slot; /* found */
 		}
-- 
1.7.5.4

